panels/applications/cc-applications-panel.c

```diff
@@ -1405,6 +1405,19 @@ update_panel (CcApplicationsPanel *self,
   }
 }
 
+/* Pseudo-applications from eos-application-tools open the app center if the
+ * app is not installed, or forward to the app if it is. Fetch the key which
+ * indicates which app replaces @info, if any, or %NULL otherwise.
+ */
+static char *
+get_replaced_by (GAppInfo *info)
+{
+  if (G_IS_DESKTOP_APP_INFO (info))
+    return g_desktop_app_info_get_string (G_DESKTOP_APP_INFO (info), "X-Endless-Replaced-By");
+
+  return NULL;
```

Comment:
Do you plan to use this function somewhere else? If not, I think returning a boolean and inverting the checked condition makes more sense. Something like:

```c
static gboolean
get_replaced_by (GAppInfo *info)
{
  g_autofree gchar *replaced_by = NULL;

  if (!G_IS_DESKTOP_APP_INFO (info))
    return FALSE;

  replaced_by = g_desktop_app_info_get_string (G_DESKTOP_APP_INFO (info), "X-Endless-Replaced-By");
  return replaced_by != NULL;
}
```

Reply:
Yep, that's a good idea. I used a different name and slightly different (but equivalent) control flow.